src/lib/utils/meta_tables/meta_system_information_table.cpp

```diff
@@ -0,0 +1,91 @@
+#ifdef __linux__
+
+#include <numa.h>
+#include <sys/sysinfo.h>
+#include <fstream>
+
+#endif
+
+#include "meta_system_information_table.hpp"
+
+namespace opossum {
+
+MetaSystemInformationTable::MetaSystemInformationTable()
+    : AbstractMetaSystemTable(TableColumnDefinitions{{"cpu_count", DataType::Int, false},
+                                                     {"system_memory_total_bytes", DataType::Long, false},
+                                                     {"numa_cpu_count", DataType::Int, false},
+                                                     {"cpu_model", DataType::String, false}}) {}
+
+const std::string& MetaSystemInformationTable::name() const {
+  static const auto name = std::string{"system_information"};
+  return name;
+}
+
+std::shared_ptr<Table> MetaSystemInformationTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  const auto cpus = _get_cpu_count();
+
+  auto numa_cpus = cpus;
+#ifdef __linux__
+  if (numa_available() != -1) {
+    numa_cpus = numa_num_task_cpus();
+  }
+#endif
+
+  int64_t ram;

```

Comment:
this is also due to the fact that Long values in a table must be signed. otherwise we would need another conversion before placing the value in the table.